phantombuster/web-browser-extension
# feat: reskin web extension

## 📝 What is this PR

[FE-71](https://phantombuster.atlassian.net/jira/software/projects/FE/boards/46?selectedIssue=FE-71)

## 🏭 How does it work
I just followed what is recommended in google [documentation](https://developer.chrome.com/docs/extensions/mv2/manifest/icons/) in order to update the web extension icons.

For the injected `Connect to something` button, please note that the extension code should inject the button in three different pages:
- Agent setup page for multistep setup scripts (new scripts), the button injection is handled by [this class](https://github.com/phantombuster/web-browser-extension/blob/7e565d5c3618aadcddef133a3716dcb62fd70cea/src/contentscript/handlers/phantombusterNewSetup.ts#L26).
- Agent setup page for non-multistep setup scripts (old scripts), the button injection is handled by [this class](https://github.com/phantombuster/web-browser-extension/blob/7e565d5c3618aadcddef133a3716dcb62fd70cea/src/contentscript/handlers/phantombusterOldSetup.ts#L23).
- Zapier editor page, the button injection is handled by [this class](https://github.com/phantombuster/web-browser-extension/blob/7e565d5c3618aadcddef133a3716dcb62fd70cea/src/contentscript/handlers/zapierCustomizeLaunch.ts#L28).

Reskinning the three buttons is something that is not achievable unfortunately because 2 uses cases cannot be reproduced and are not testable at all. It seems that the extension is not working with zapier anymore and that there is no more non-multisteps scripts that have a session cookie field. Please check this [slack thread](https://thephantomcompany.slack.com/archives/C0232FW7BFS/p1652706658183269).

For the operational use case (multistep setup scripts), I removed the pb2-css classes references and defined the button styling within the extension code.

## 🧪 How to test

### First, build and load the extension into your browser

- run `nvm use` on the parent folder of the web extension
- run `npm i`
- run `npm run build`
- Follow [these steps](https://developer.chrome.com/docs/extensions/mv3/getstarted/#unpacked) to load an unpacked extension into chrome.
- Follow [these steps](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Your_first_WebExtension#installing) to load the extension into firefox.

### Testing the extension
- Check the new icon of the extension
- Go to phantombuster.com, use different scripts (twitter, facebook, linkedin ...), and check the new styling of the `Connect to {service}` button.
- Check that the extension is still working properly.

**Note:** You cant' use the web extension with our deploy previews. The list of hosts that the extension will work on is [hardcoded](https://github.com/phantombuster/web-browser-extension/blob/7e565d5c3618aadcddef133a3716dcb62fd70cea/src/contentscript/handlers/phantombuster.ts#L5) and doesn't include `**-buster.preview.phantomcompany.io`.
